terraform/graph_config_node_resource.go

```diff
@@ -249,6 +249,38 @@ func (n *GraphNodeConfigResource) DestroyNode(mode GraphNodeDestroyMode) GraphNo
 	return result
 }
 
+// GraphNodeNoopPrunable
+func (n *GraphNodeConfigResource) Noop(opts *NoopOpts) bool {
+	// We don't have any noop optimizations for destroy nodes yet
+	if n.DestroyMode != DestroyNone {
+		return false
+	}
+
+	// If there is no diff, then we aren't a noop
+	if opts.Diff == nil || opts.Diff.Empty() {
+		return false
+	}
+
+	// If we have no module diff, we're certainly a noop
+	if opts.ModDiff == nil || opts.ModDiff.Empty() {
+		return true
+	}
```

Comment:
These two clauses threw me off. Why do we know that an empty `Diff` is definitely not a noop but an empty `ModDiff` is definitely a noop?